src/shim.ts

```diff
@@ -6,43 +26,211 @@ import {
+ *     // orchestrator body
+ * });
+ * ```
+ */
+export function orchestration(functionName: string, handler: OrchestrationHandler): void {
+    app.generic(functionName, {
+        trigger: trigger.orchestration(),
+        handler: createOrchestrator(handler),
+    });
+}
+
+/**
+ * Registers a function as a Durable Entity for your Function App.
+ *
+ * @param functionName the name of your new durable entity
+ * @param handler the function that should act as an entity
+ *
+ * @example Register a counter entity
+ * ```javascript
+ * const df = require("durable-functions");
+ *
+ * df.entity('CounterEntity', function (context) {
+ *     // entity body
+ * });
+ * ```
+ */
+export function entity<T = unknown>(functionName: string, handler: EntityHandler<T>): void {
+    app.generic(functionName, {
+        trigger: trigger.entity(),
+        handler: createEntityFunction(handler),
+    });
+}
+
+/**
+ * Registers a function as an Activity Function for your Function App
+ *
+ * @param functionName the name of your new activity function
+ * @param options the configuration options for this activity,
+ * specifying the handler and the inputs and outputs
+ *
+ * @example Register an activity function with extra inputs and outputs
+ * ```javascript
+ * const df = require("durable-functions");
+ * const { input, output } = require("@azure/functions");
+ *
+ * df.activity('MyActivity', {
+ *   extraInputs: input.storageBlob({ path: 'test/{test}', connection: 'conn' }),
+ *   extraOutputs: output.storageBlob({ path: 'test/{test}', connection: 'conn' }),
+ *   handler: function (context) {
+ *      // activity body
+ *   }
+ * });
+ * ```
+ */
+export function activity<T = unknown>(functionName: string, options: ActivityOptions<T>): void {
+    app.generic(functionName, {
+        trigger: trigger.activity(),
+        ...options,
+    });
+}
+
+/**
+ * Registers a function as a Durable Client for your Function App,
+ * allowing extra configurations, such as extra inputs and outputs
+ *
+ * @param functionName the name of your new Durable Client function
+ * @param options object specifying configuration options,
+ * such as handler, inputs and outputs
+ *
+ */
+export function client(functionName: string, options: DurableClientOptions): void {
+    const clientInput: DurableClientInput = input.client();
+    const clientHandler: DurableClientHandler = options.handler;
+
+    if (options.extraInputs) {
+        options.extraInputs.push(clientInput);
+    } else {
+        options.extraInputs = [clientInput];
+    }
+
+    const handler = (context: InvocationContext, triggerData: unknown) => {
+        const client = getClient(context, clientInput);
+        return clientHandler(context, triggerData, client);
+    };
+
+    app.generic(functionName, {
+        ...options,
+        handler,
+    });
+}
+
+/**
+ * The root namespace to help create durable trigger configurations
+ */
+export namespace trigger {

```

Comment:
It didn't seem like it made much sense to have a separate `app` namespace for Durable, since the methods that would be there should be the only main methods that the user interacts with to begin with. Conversely, I don't see `df.trigger` and `df.input()` being used outside of `app.generic()` use-cases, which aren't nearly as common. Another reason against it is that it avoids duplication of namespaces for most use-cases. 

In terms of duplicating namespaces, I would be more in favor of recommending the first pattern over the second. I don't think the SDK (a separate package) should be in the business of avoiding the most natural names for our namespaces to accommodate `@azure/functions` or any other package.

I could also see a third pattern being used:

```TS
import { input as azFuncApp } from '@azure/functions';
import { input as dfInput } from 'durable-functions';
```

Or they could choose to keep one as just `input` and rename just the other

Reply:
`@azure/functions` isn't just "any other package" though. It's the parent package that everything in durable-functions is based off of. It feels like this should be a common enough pattern where one package is the clear parent of another, which is why I was hoping we could find other examples instead of designing a pattern from scratch. For example, I would think there are many packages that extend the express or fastify frameworks

When looking for examples, please keep in mind that we can totally change `@azure/functions` if necessary (of course, in a separate PR). You are serving as a trailblazer for people using/extending the package and I expect some bumps.

Reply:
If we're looking for precent: In PyStein, we're exporting the **entire** API (triggers and input bindings) through a `dfApp` namespace. Personally, I like it as it keeps the Function-declaration methods all in the same place and it also mimicks the baseline PyStein code organization

Reply:
So I guess there are two different questions here:

1. Should APIs for registering functions be included in some kind of `app` namespace or in the global DF package directly?
2. Should DF namespaces differentiate themselves from `@azure/functions` namespaces by including `df` prefix? 

For those two questions, my personal preference is basically what I have in this PR: APIs for registering functions should be accessed by `df.` directly, and no to the second question. But I don't have strong feelings about this. Do you folks have strong feelings one way or the other?

Reply:
I don't fully understand the subtlety between questions (1) and (2). Aren't these just two different ways of importing a namespace? @hossam-nasr?

Reply:
Okay, I've added the `app` namespace, but kept the names as-is for now. I'm happy to rename them to `dfApp`, `dfInput`, and `dfTrigger` (or similar) if we get strong feedback for this (or if anyone has strong feelings about this)